File: Src/ButterflyNet.Parser/Strategies/DefinitionListStrategy.cs
using ButterflyNet.Parser.Satisfiers;
using ButterflyNet.Parser.Scopes;

namespace ButterflyNet.Parser.Strategies {
	public class DefinitionListStrategy : ScopeDrivenStrategy, ITokenProvider {
		public DefinitionListStrategy() {
			AddSatisfier<StartOfLineSatisfier>();
			AddSatisfier(new NegatingSatisfier(new InScopeStackSatisfier(ScopeTypeCache.DefinitionTerm)));
		}

		protected override void DoExecute(ParseContext context) {
			if (!context.Scopes.ContainsType(ScopeTypeCache.DefinitionList)) {
				OpenScope(new DefinitionListScope(), context);
			}

			OpenScope(new DefinitionTermScope(), context);
		}

		public string Token { get { return ";"; } }
	}
}

File: src/js/src/strategies/DefinitionListStrategy.js
function DefinitionListStrategy() {
	DefinitionListStrategy.$parent.call(this);
	
	this.addSatisfier(new StartOfLineSatisfier());
	this.addSatisfier(new CannotNestInsideInlineSatisfier());
	this.addSatisfier(new NegatingSatisfier(new InScopeStackSatisfier(ScopeTypeCache.definitionTerm)));
	this.setAsTokenTransformer(";");
	
	this.doExecute = function(context) {
		if (!context.scopes.containsType(ScopeTypeCache.definitionList)) {
			this.openScope(new DefinitionListScope(), context);
		}
		
		this.openScope(new DefinitionTermScope(), context);
	};
}

extend(ScopeDrivenStrategy, DefinitionListStrategy);